File: src/docs/Pages/TextAlign.fs
module Docs.Pages.TextAlign

open Feliz
open Elmish
open Feliz.UseElmish
open Feliz.DaisyUI
open Feliz.Tailwind
open Feliz.Tailwind.Operators
open Docs.SharedView


let textAlignStyles = [
    textAlign.left
    textAlign.center
    textAlign.right
    textAlign.justify
    textAlign.start
    textAlign.``end``
]


let str prop =
    if prop = textAlign.left then "textAlign.left"
    elif prop = textAlign.center then "textAlign.center"
    elif prop = textAlign.right then "textAlign.right"
    elif prop = textAlign.justify then "textAlign.justify"
    elif prop = textAlign.start then "textAlign.start"
    elif prop = textAlign.``end`` then "textAlign.end"
    else failwith "Unknown TextAlign property"

let renderTextAlign textAlign =
    let example =
        Html.p [
            textAlign
            prop.text
                "So I started to walk into the water. I won't lie to you boys, I was terrified. But I pressed on, and as I made my way past the breakers a strange calm came over me. I don't know if it was divine intervention or the kinship of all living things but I tell you Jerry at that moment, I was a marine biologist."
        ]

    let code =
        $"""Html.p [
            {str textAlign}
            prop.text "So I started to walk into the water. I won't lie to you"""

    let title =
        Html.text
            "Use of TextAlign is straightforward. Just add the TextAlign property to the element you want to align. Here are some examples:"

    codedView title code example

[<ReactComponent>]
let TextAlignView () =
    React.fragment [
        for textAlign in textAlignStyles do
            renderTextAlign textAlign
    ]

File: src/docs/output/Pages/TextAlign.js
import { ofArray } from "../fable_modules/fable-library-js.4.24.0/List.js";
import { createObj, equals } from "../fable_modules/fable-library-js.4.24.0/Util.js";
import { createElement } from "react";
import React from "react";
import * as react from "react";
import { codedView } from "../SharedView.js";
import { map, delay, toList } from "../fable_modules/fable-library-js.4.24.0/Seq.js";

export const textAlignStyles = ofArray([["className", "text-left"], ["className", "text-center"], ["className", "text-right"], ["className", "text-justify"], ["className", "text-start"], ["className", "text-end"]]);

export function str(prop) {
    if (equals(prop, ["className", "text-left"])) {
        return "textAlign.left";
    }
    else if (equals(prop, ["className", "text-center"])) {
        return "textAlign.center";
    }
    else if (equals(prop, ["className", "text-right"])) {
        return "textAlign.right";
    }
    else if (equals(prop, ["className", "text-justify"])) {
        return "textAlign.justify";
    }
    else if (equals(prop, ["className", "text-start"])) {
        return "textAlign.start";
    }
    else if (equals(prop, ["className", "text-end"])) {
        return "textAlign.end";
    }
    else {
        throw new Error("Unknown TextAlign property");
    }
}

export function renderTextAlign(textAlign) {
    let value;
    const example = createElement("p", createObj(ofArray([textAlign, (value = "So I started to walk into the water. I won\'t lie to you boys, I was terrified. But I pressed on, and as I made my way past the breakers a strange calm came over me. I don\'t know if it was divine intervention or the kinship of all living things but I tell you Jerry at that moment, I was a marine biologist.", ["children", value])])));
    const code = `Html.p [
            ${str(textAlign)}
            prop.text "So I started to walk into the water. I won't lie to you`;
    let title;
    const value_2 = "Use of TextAlign is straightforward. Just add the TextAlign property to the element you want to align. Here are some examples:";
    title = value_2;
    return codedView(title, code, example);
}

export function TextAlignView() {
    const xs = toList(delay(() => map(renderTextAlign, textAlignStyles)));
    return react.createElement(react.Fragment, {}, ...xs);
}

